test/e2e/apimachinery/openapiv3.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/onsi/ginkgo/v2"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apiextensions-apiserver/test/integration/fixtures"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/openapi3"
+	aggregatorclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	"k8s.io/kube-openapi/pkg/spec3"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	// ensure libs have a chance to initialize
+	_ "github.com/stretchr/testify/assert"
+)
+
+var _ = SIGDescribe("OpenAPIV3", func() {
+	f := framework.NewDefaultFramework("openapiv3")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+
+	/*
+		Release : v1.27
+		Testname: OpenAPI V3 RoundTrip
+		Description: Fetch the OpenAPI v3 of all group versions. The OpenAPI specs MUST roundtrip successfully.
+	*/
+	ginkgo.It("should round trip OpenAPI V3 for all group versions", func(ctx context.Context) {
+		c := openapi3.NewRoot(f.ClientSet.Discovery().OpenAPIV3())
+		gvs, err := c.GroupVersions()
+		framework.ExpectNoError(err)
+		for _, gv := range gvs {
+			spec1, err := c.GVSpec(gv)
+			framework.ExpectNoError(err)
+			specMarshalled, err := json.Marshal(spec1)
+			framework.ExpectNoError(err)
+			var spec2 spec3.OpenAPI
+			json.Unmarshal(specMarshalled, &spec2)
+
+			if !reflect.DeepEqual(*spec1, spec2) {
+				diff := cmp.Diff(*spec1, spec2)
+				framework.Failf("%s", diff)
+			}
+		}
+	})
+
+	/*
+		Release : v1.27
+		Testname: OpenAPI V3 CustomResourceDefinition
+		Description: Create a CustomResourceDefinition. The OpenAPI V3 document of the CustomResourceDefinition MUST be created. The OpenAPI V3 MUST be round trippable.
+	*/
+	ginkgo.It("should publish OpenAPI V3 for CustomResourceDefinition", func(ctx context.Context) {
```

Comment:
Is it possible to combine these tests? It seems unnecessarily resource intensive to set up separate extensions for each of them.

Reply:
Discussed offline, the tests do not need to be combined as the sample-apiserver is only started once in the test suite.